### PR TITLE
Allow getting/setting a component in a single update

### PR DESCRIPTION
**Pull request by cvincent**
Calling `update/2` with a function allows getting and setting a component in one call, like so:

```elixir
MyComponent.update(123, fn (old_val) -> old_val + 1 end)
```

### CODE REVIEW

**Comment by APB9785** [comment]
Hi Chris 👋  thanks for your interest in and contribution to ECSx!  This idea (updating Component API to include passing a function to `update/2`) has been investigated in the past, and we eventually decided against it.  I'll share the reasoning in hopes that it helps you find the best path forward for your use-case(s).

First, let me refactor your code to avoid `ECSx.Base`:
```elixir
def update(entity_id, fun) when is_function(fun, 1) do
  old_value = get_one(entity_id)
  new_value = fun.(old_value)
  update(entity_id, new_value)
end
```
This is important because `update/2` performs type-checking to ensure the new_value is appropriate, whereas `ECSx.Base.update/4` does not.  I also removed your return value, as the current pattern throughout the entire API is that write operations return `:ok`.

With this example in mind, let's take a look at a potential use-case inside a System.

Standard approach:
```elixir
  old_hp = HitPoints.get_one(entity_id)
  new_hp = decrement(old_hp)
  HitPoints.update(entity_id, new_hp)
```

Update with function:
```elixir
HitPoints.update(entity_id, &decrement/1)
```

Here we can see the trade-off clearly - the former example makes `old_hp` and `new_hp` available for use later, at the cost of two additional lines of code.  The latter example uses less code, but you do not get access to the HP values.

The design of ECSx is intended to favor the former approach, which is broad and covers more use-cases, while still allowing the extensibility of the Component API to handle more specific approaches for individual use-cases.  If you know that the HitPoints component will be updated via function without concern for the actual values, you can add this to the `MyApp.Components.HitPoints` module:
```elixir
defmodule MyApp.Components.HitPoints do
  ...
  def update(entity_id, fun) when is_function(fun, 1) do
    old_value = get_one(entity_id)
    new_value = fun.(old_value)
    update(entity_id, new_value)
  end
end
```
And now the desired functionality is available to your throughout your codebase!

In summary, we provide the fundamental "building blocks" in the Component API, which every app will need.  Then users can compose those blocks together to make more specific functions if/when the need arises (if the same code pattern is being repeated in several locations).

Hope this helps! I'm happy to receive any feedback on our design, or answer any questions you might have.  Thanks again for your time and interest 🙂

**Comment by cvincent** [comment]
Sounds good, thanks for the thorough explanation! Wishing this project success, and excited to see what people build with it.
